extension/packagemanagerextension/nbextension/methods.js

```diff
@@ -0,0 +1,146 @@
+                    element = element.split('=');
+                    let pkg = element[0];
+                    let ver = element[1];
+                    let nver = checknewversion(updates, pkg, ver);
+                    if (nver != ver) {
+                        packages.push(pkg);
+                        html += "<li>";
+                        html += pkg + "    " + ver + " -> " + nver;
+                        html += "</li>";
+                        ct += 1;
+                    }
+                });
+                html += "</ul>";
+            }
+            else {
+                let list = $('.installed-values');
+                html += "<ul>";
+                for (let item of list) {
+                    let pkg = $(item).find('.value-name')[0].innerText;
+                    let ver = $(item).find('.value-version')[0].innerText;
+                    let nver = checknewversion(updates, pkg, ver);
+                    if (nver != ver) {
+                        packages.push(pkg);
+                        html += "<li>";
+                        html += pkg + "    " + ver + " -> " + nver;
+                        html += "</li>";
+                        ct += 1;
+                    }
+                }
+                html += "</ul>";
+            }
+            if (ct === 0) {
+                html = "<p> There are no updates available </p>"
+            }
+            common.confirm("Update Packages", $.parseHTML(html), "Confirm", function () {
+                api.update_packages(packages, project);
+            });
+        });
+    }
+
+
+    /*
+    This function lists down all the packages selected for removal and requires a user confirmation.
+    */
+
+    function deletepkg() {
+        $('#deletebtn').unbind();
+        $('#deletebtn').click(function () {
+            let selectedPackages = common.get_selected_packages();
+            let url = location.href.split('/');
+            let project = 'SWAN_projects/' + url[url.length - 1];
+            let html = "<p> The following packages are about to be deleted: </p> </br>";
+            html += "<ul>";
+            let packages = [];
+            selectedPackages.forEach(element => {
+                element = element.split('=')[0];
+                packages.push(element);
+                html += "<li>";
+                html += element;
+                html += "</li>";
+            });
+            html += "</ul>";
+            if (packages.length === 0)
+                html = "<p> No packages selected </p>";
+            common.confirm("Delete Packages", $.parseHTML(html), "Confirm", function () {
+                api.delete_packages(packages, project);
+            });
+        });
+    }
+
+
+    /*
+    This function lists down all the packages selected for installation and requires a user confirmation.
+    */
+
+    function installpkg() {
+        $('#installbtn').unbind();
+        $('#installbtn').click(function () {
```

Comment:
Do you see the pattern here? Isn't this exactly the same code as deletepkg? (which should be "delete_package" to be consistent, btw)

Reply:
Did you read my comment? Why did you mark it as resolved if you didn't address the issue??

Reply:
I can accept that these functions are kept as they are IF you move all the html code to the `display_msg`